src/filewatch.erl

```diff
@@ -22,40 +53,38 @@ priv_dir() ->
 
 %% If you put this in on_load, it will immediately get unloaded when that child dies.
 load() ->
-    case erl_ddll:load_driver(priv_dir(), "filewatch") of
+    case erl_ddll:load_driver(priv_dir(), ?DRIVER_NAME) of
         ok -> ok;
         {error, already_loaded} -> ok;
         {error, Message} ->
-            error_logger:error_msg("filewatch: error loading driver: ~p",
-                                   [erl_ddll:format_error(Message)])
+            error_logger:error_msg("~p: error loading driver: ~p",
+                                   [?DRIVER_NAME,
+                                    erl_ddll:format_error(Message)])
     end.
 
 
--spec start(pid(), [{file:name_all(), term()}]) -> {ok, handle()} | {error, _}.
-
-start(Self, Pairs) ->
-    %% CooldownMs = case application:get_env(cooldown) of
-    %%                  {ok, Cd} -> Cd;
-    %%                  undefined -> 3000
-    %%              end,
-    Pid = spawn_link(fun () -> init(Self, Pairs) end),
-    {ok, Pid}.
-
--spec stop(handle()) -> ok | {error, _}.
-
-stop(Handle) ->
-    Handle ! terminate,
-    ok.
 
 init(Pid, Pairs) ->
     ok = load(),
-    Port = open_port({spawn_driver, "filewatch "}, [in]),
+    Arg = io_lib:format("~p ~p", [?DRIVER_NAME, ?COOLDOWN_MS]),
+    Port = open_port({spawn_driver, Arg}, [in]),
     WatchMap = create_watch_map(Pairs, Port, maps:new()),
-    watch(#state{pid=Pid, port=Port, map=WatchMap}).
+    [cast(Pid, Term) || {_, Term} <- Pairs],
+    watch(#state{pid = Pid, port = Port, watch_map = WatchMap}).
 
 
 create_watch_map([], _Port, Map) -> Map;
 create_watch_map([{Path, Term} | Rest], Port, Map) ->
+    NewMap = case filelib:is_file(Path) of
+                 true -> add_to_watch_map({Path, Term}, Port, Map);
+                 false -> error_logger:error_msg("~p: Path ~p is not a file",
```

Comment:
Good point. Rather then ensuring that every file exists, it's enough to verify that every parent directory for every file exists, since that's what is actually getting watched by inotify.